prototype/link/scripts/Analysis_Timing_functions/EyeTracking_Experiment_MM.m

```diff
@@ -0,0 +1,300 @@
+EyeData = varargin{1};
+Data = varargin{2};
+GenerateTrials = varargin{3};
+
+TR = 2; % Assume the TR is 2s
+
+is_conservative = 0; % If this is 1 then you will only consider a present response acceptable. If it is set to zero then only a response that isn't off screen is acceptable. So if there is no eye tracking data or there is
+
+% Specify the experiment ID
+Experiment = 'MM';
+
+%If it wasn't quit out
+
+for IdxCounter=1:length(EyeData.Timecourse.(Experiment))
+    %What block and repetition does this Idx correspond to
+    Idx_Name=EyeData.Idx_Names.(Experiment)(IdxCounter,:);
+    Block_Name=sprintf('Block_%d_%d', Idx_Name(1), Idx_Name(2));
+    MovieName = GenerateTrials.Experiment_MM.Parameters.BlockNames{Idx_Name(1)};
+
+    %Extract the timecourse
+    Timecourse=EyeData.Timecourse.(Experiment){IdxCounter};
+    
+    % Get the timing of each eye tracker frame
+    FrameTiming=EyeData.Timing.(Experiment){IdxCounter}(:,1);
+    
+    % Get the time stamps
+    
+    MovieStart = Data.Experiment_MM.(Block_Name).Timing.Movie_1.movieStart.Local;
+    MovieEnd = Data.Experiment_MM.(Block_Name).Timing.Movie_1.movieEnd.Local;
+    
+    % What time points do you have measurements for
+    time_points = FrameTiming ~= 0;
+    
+    % When did the block start
+    onset_time = FrameTiming(find(time_points == 1, 1));
+    
+    % Elapsed time for each eye frame
+    elapsed_time = (FrameTiming(time_points) - onset_time) / EyeData.EyeTrackerTime.slope;
+    
+    % Pull out the block's gaze data
+    gaze_data_block = Timecourse(time_points);
+    
+    % Remap the data to make it easier to read
+    gaze_data_block(gaze_data_block==0) = 7; % Turn all off screen responses into a different response
+    gaze_data_block = 7 - gaze_data_block; % Now subtract 7 so that the lowest number (which was a 5 for present) becomes a 2
+    
+    % Get the block start and stop time in eye tracker time
+    block_start = (MovieStart * EyeData.EyeTrackerTime.slope) + EyeData.EyeTrackerTime.intercept;
+    block_end = (MovieEnd * EyeData.EyeTrackerTime.slope) + EyeData.EyeTrackerTime.intercept;
+    
+    % Specify the critical window for doing exclusions based on
+    % the movie timing
+    if sum((FrameTiming(time_points) > block_start) .* (FrameTiming(time_points) < block_end)) > 0
+        [~,block_start_idx]=min(abs(FrameTiming - block_start));
+        [~,block_end_idx]=min(abs(FrameTiming - block_end));
+        Exclude.CriticalWindow{IdxCounter}=block_start_idx:block_end_idx; %In terms of the indexes in the time course data
+    else
+        % No data was found in this critical window so specify that here
+        Exclude.CriticalWindow{IdxCounter}=[];
+    end
+    
+    % Turn these times in to seconds
+    block_start_elapsed = (block_start - onset_time) / EyeData.EyeTrackerTime.slope;
+    block_end_elapsed = (block_end - onset_time) / EyeData.EyeTrackerTime.slope;
+    
+    % Specify all of the Drop periods
+    Drop_windows = [];
```

Comment:
Might be worth putting a warning in the comments here that the timing was hard coded